.buildkite/pipeline.yml

```diff
@@ -1,2 +1,65 @@
+agents:
+  image: "docker.elastic.co/ci-agent-images/eck-region/go-builder-buildkite-agent:1.20"
+  cpu: "4"
+  memory: "4G"
+
+env:
+  DOCKER_IMAGE: "docker.elastic.co/elasticsearch-k8s-metrics-adapter/elasticsearch-metrics-adapter"
+  DOCKER_IMAGE_TAG: "git-${BUILDKITE_COMMIT:0:12}"
+
 steps:
-  - command: echo "🐙"'
\ No newline at end of file
+  - label: ":go: Run unit tests"
+    command: "make test"
+    key: "tests"
+    artifact_paths:
+      - "coverage.out"
+
+  - label: ":sonarqube: Static Code Analysis"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/serverless/shared-analysis-token"
+    agents:
+      image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+    command:
+      - "buildkite-agent artifact download coverage.out ."
+      - "/scan-source-code.sh"
+    soft_fail: true
+    depends_on: "tests"
+
+  - label: ":go: Build"
+    command: "make all"
+  
+  # Validate the Helm charts
+  - label: ":helm: validate helm charts"
+    command: "make validate-helm"
+  
+  - label: "Package Helm Charts and push into registry"
+    command: "make -C /agent helm-publish"
+    env:
+      OCI_REGISTRY_PATH: oci://${DOCKER_IMAGE}
+      CHART_APP_VERSION: "${DOCKER_IMAGE_TAG}" 
+    agents:
+      image: "docker.elastic.co/ci-agent-images/serverless-helm-builder:0.0.2"
+
+  - wait
+
+  # multi-arch container image build
+  - group: ":docker: Build Container Images"
+    steps:
+      - label: ":docker: :seedling: Trigger Image Creation"
+        command: "make -C /agent generate-docker-images"
+        agents:
+          image: "docker.elastic.co/ci-agent-images/serverless-docker-builder:0.0.4"
```

Comment:
So am I right in reading that this will build/push a docker/container image on every commit to a PR? It seems as though we'd only build an image on merge to main? Or git tag?

Reply:
it would build/push a docker container image on every commit only if the PR is not from a fork, otherwise it build on merge to main on creation of a tag